### PR TITLE
Add code to skip tests when conda-env isn't on the PATH

### PR DESCRIPTION
**Pull request by tswicegood**
This should fix the tests on TravisCI

### CODE REVIEW

**Comment by tswicegood** [comment]
Going to let Travis do it's think, then merge if this passes.

File: tests/test_activate.py [hunk 1]
```diff
@@ -73,6 +87,7 @@ def _write_entry_points(envs):
 mkdir -p {envs}/test2/bin
 """
 
+@unittest.skipIf(skip_tests, "conda-env not installed on path")
```
Comment (asmeurer) [inline tests/test_activate.py]: Actually, I'd rather they failed. I don't see why conda-env wouldn't be installed as it's a dependency of conda, and if these tests are silently skipped, then we won't know if something in activate breaks.

File: tests/test_activate.py [hunk 2]
```diff
@@ -21,7 +23,19 @@
         if not stderr:
             shells.append(shell)
 
+    # activate and deactivate are no longer part conda, so we can't copy them
+    # from the source tree.  They should normally be installed, so this pulls
+    # them from the path.
```
Comment (asmeurer) [inline tests/test_activate.py]: I see. Really, these tests should be moved to conda-env.

**Comment by asmeurer** [comment]
Doesn't Travis CI install conda-env before testing?

**Comment by tswicegood** [comment]
@asmeurer yes, but these tests shouldn't fail because of a missing dependency.  Consider the case where you check out this code on a machine without conda installed.  You should be able to run the unit tests and only have failures for code that actually failed.  When a dependency isn't met, it should skip with a note as to why it skipped (thus the message about conda-env not being on the path).

I'm with you that these should be moved to conda-env, but they need to be refactored.  Since they serve as a nice early warning system, I'm doing the least I can to get them back to passing and leaving that as an exercise for another day.

**Comment by asmeurer** [comment]
> @asmeurer yes, but these tests shouldn't fail because of a missing dependency. Consider the case where you check out this code on a machine without conda installed. You should be able to run the unit tests and only have failures for code that actually failed. When a dependency isn't met, it should skip with a note as to why it skipped (thus the message about conda-env not being on the path).

I agree with that for optional dependencies, but conda-env is a hard dependency. 

> I'm with you that these should be moved to conda-env, but they need to be refactored. Since they serve as a nice early warning system, I'm doing the least I can to get them back to passing and leaving that as an exercise for another day.

I know, you can't touch code without refactoring it.

**Comment by tswicegood** [comment]
Code passed on TravisCI.

**Comment by asmeurer** [comment]
Thanks for fixing this.

**Pull request merged**